Source/Peripheral+Convenience.swift

```diff
@@ -78,18 +82,20 @@ extension Peripheral {
      - Returns: Observation which emits `Next` event, when specified descriptor has been found.
      Immediately after that `.Complete` is emitted.
      */
-    public func descriptor(with identifier: DescriptorIdentifier) -> Observable<Descriptor> {
+    public func descriptor(with identifier: DescriptorIdentifier) -> Single<Descriptor> {
         return .deferred {
             return self.characteristic(with: identifier.characteristic)
-                .flatMap { characteristic -> Observable<Descriptor> in
+                .flatMap { characteristic -> Single<Descriptor> in
                     if let descriptors = characteristic.descriptors,
                         let descriptor = descriptors.first(where: { $0.uuid == identifier.uuid }) {
                         return .just(descriptor)
                     }
                     return characteristic.discoverDescriptors()
+                        .asObservable()
```

Comment:
Why do we need `asObservable()` here?

Reply:
This can be replace with something like this:
```
return self.discoverServices([identifier.uuid])
                    .map {
                        if let service = $0.first {
                            return service
                        }
                        throw RxError.noElements
                    }
```

or changed to a `Maybe`.
I am not sure what the expected behaviour is, when the Service UUID is not present in the Peripheral. Should it throw an error (`Single`) or just complete (`Maybe`)?

Reply:
@Kacper20 @Cierpliwy Any suggestions regarding this?

Reply:
We're gonna discuss it today as it touches the expected behavior and we don't want to break logic for clients of the library. Thanks for the patience! :)